x-pack/plugins/lens/public/app_plugin/time_range.ts

```diff
@@ -43,10 +42,10 @@ export function useTimeRange(
       to,
     });
     return { fromDate: min?.toISOString() || from, toDate: max?.toISOString() || to };
-    // recalculate current date range if current "now" value changes because calculateBounds
-    // depends on it internally
+    // recalculate current date range if the session gets updated because it
+    // might change "now" and calculateBounds depends on it internally
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [timefilter, currentNow, from, to]);
+  }, [timefilter, data.search.session.getSessionId(), from, to]);
```

Comment:
nit: I'd pass `state.sessionId` from the app component here instead. 
Not sure if there is a condition when those can get desynced, but passing it down seems simpler to follow.